src/org/rascalmpl/parser/uptr/recovery/LiteralMatcher.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2022, NWO-I Centrum Wiskunde & Informatica (CWI)
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **/
+
+ package org.rascalmpl.parser.uptr.recovery;
+
+public class LiteralMatcher implements InputMatcher {
+    private int[] chars;
+
+    public LiteralMatcher(String literal) {
+        chars = new int[literal.length()];
+        for (int i=0; i<literal.length(); i++) {
+            chars[i] = literal.codePointAt(i);
+        }
+    }
+
+    public LiteralMatcher(int[] literal) {
+        chars = literal;
+    }
+
+    @Override
+    public MatchResult findMatch(int[] input, int startLocation) {
```

Comment:
It might be possible but `LiteralMatcher` is used most often and should be fast so a special implementation for this situation seems prudent. I have not profiled this so it could be argued this is a case of premature optimization ;-)